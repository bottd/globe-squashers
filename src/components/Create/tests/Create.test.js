import React from 'react';
import {shallow, mount} from 'enzyme';
import {Create} from '../';
import {render} from 'react-testing-library';

describe('Create', () => {
  describe('Create component', () => {
    let wrapper;
    beforeEach(() => {
     wrapper = shallow(
        <Create
          addProof={jest.fn()}
          fetchImages={jest.fn()}
          fetchImage={jest.fn()}
          setPage={jest.fn()}
          resetImages={jest.fn()}
          page={2}
          images={[1, 2, 3]}
          image={{}}
        />,
      );
    });

    it('Should render like snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('Should have be loaded after mount', () => {
      expect(wrapper.state().isLoaded).toBe(true);
    });

    it('Should be able to cycleImages', async () => {
      await wrapper.instance().cycleImage(1);
      expect(wrapper.state().currentImage).toBe(1);
    });
  });
});
