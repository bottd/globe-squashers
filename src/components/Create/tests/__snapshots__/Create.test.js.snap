// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Create component Should render like snapshot 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Create
    fetchImage={[MockFunction]}
    fetchImages={[MockFunction]}
    image={Object {}}
    images={Array []}
    page={2}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "fetchImage": [MockFunction],
      "fetchImages": [MockFunction],
      "image": Object {},
      "images": Array [],
      "page": 2,
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="Create"
      >
        <form>
          <textarea
            class="note-input"
            placeholder="Write your note here"
            rows="5"
          />
        </form>
        <img
          class="hubble-img"
        />
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form>
            <textarea
              className="note-input"
              onChange={[Function]}
              placeholder="Write your note here"
              rows="5"
              value=""
            />
          </form>,
          <img
            className="hubble-img"
          />,
        ],
        "className": "Create",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <form>
            <textarea
              class="note-input"
              placeholder="Write your note here"
              rows="5"
            />
          </form>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <textarea
              className="note-input"
              onChange={[Function]}
              placeholder="Write your note here"
              rows="5"
              value=""
            />,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <textarea
                class="note-input"
                placeholder="Write your note here"
                rows="5"
              />,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "note-input",
                "onChange": [Function],
                "placeholder": "Write your note here",
                "rows": "5",
                "value": "",
              },
              "ref": null,
              "rendered": Array [
                undefined,
              ],
              "type": "textarea",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": <img
            class="hubble-img"
          />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "hubble-img",
            "src": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "img",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "fetchImage": [MockFunction],
        "fetchImages": [MockFunction],
        "image": Object {},
        "images": Array [],
        "page": 2,
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="Create"
        >
          <form>
            <textarea
              class="note-input"
              placeholder="Write your note here"
              rows="5"
            />
          </form>
          <img
            class="hubble-img"
          />
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <form>
              <textarea
                className="note-input"
                onChange={[Function]}
                placeholder="Write your note here"
                rows="5"
                value=""
              />
            </form>,
            <img
              className="hubble-img"
            />,
          ],
          "className": "Create",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <form>
              <textarea
                class="note-input"
                placeholder="Write your note here"
                rows="5"
              />
            </form>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <textarea
                className="note-input"
                onChange={[Function]}
                placeholder="Write your note here"
                rows="5"
                value=""
              />,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <textarea
                  class="note-input"
                  placeholder="Write your note here"
                  rows="5"
                />,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "note-input",
                  "onChange": [Function],
                  "placeholder": "Write your note here",
                  "rows": "5",
                  "value": "",
                },
                "ref": null,
                "rendered": Array [
                  undefined,
                ],
                "type": "textarea",
              },
            ],
            "type": "form",
          },
          Object {
            "instance": <img
              class="hubble-img"
            />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": undefined,
              "className": "hubble-img",
              "src": undefined,
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "img",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
